modules/pxp-module-puppet

```diff
@@ -282,6 +282,10 @@ def action_run(input)
         1
       end
 
+      # flush the stdout/stderr before writing the exitcode
+      # file to avoid pxp-agent reading incomplete output
+      $stdout.fsync
```

Comment:
You're passing 5, so pxp-agent knows what to do and will use what's on stdout.
I think you should configure the un-buffered behaviour anyway for consistency.